FunctionalForm: Revise doxygen syntax

bradking opened the pull request:
The latter is easier to read in C code.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/3853)
<!-- Reviewable:end -->

bradking commented:
Assigning to @sherm1 for platform review and integration.  There is no need for feature review for this change.

sherm1 commented:
Thanks! A few meta-comments on doxygen comments, PTAL. 

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/common/functional_form.h, line 17 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYnYEXVfH45T57o1ZE:-KUYnYEYN9E5MMZ1IqpY:bpgaw2z) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L17)):*
> ```objc
> namespace drake {
> 
> /** @brief Represent an abstract form of a function of zero or more variables.
> ```

Please remove `@brief` (here and elsewhere). We switched to autobrief so Doxygen will pick up the first sentence as the brief anyway so we banned `@brief` as clutter.

---

*[drake/common/functional_form.h, line 23 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYorMqr2_9PhMRJ0MS:-KUYorMqr2_9PhMRJ0MT:b-dr0jcf) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L23)):*
> ```objc
>  * output.  The form may be one of:
>  *
>  * - @anchor zero
> ```

Doxygen documentation says: `Anchors can currently only be put into a comment block that is marked as a page (using \page) or mainpage (\mainpage).` Does that apply here? If not, do these anchors actually work?

---

*[drake/common/functional_form.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYnmH7VstFd6Kj-hqu:-KUYnmH7VstFd6Kj-hqv:b1wrzmd) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L42)):*
> ```objc
>  * - @anchor undefined
>  *   An @b undefined combination of zero or more variables
>  *   (<tt>f(x)</tt> where @c f contains an undefined operation).
> ```

BTW, some time ago Doxygen added Markdown support. That means you can use backtics for a tidier way of getting code fonts. So
```
`f(x)` and `f` work
```
Easier to read in the header file and produces the same doxygen output. Other Markdown goodies work too.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-:-KUYpvHsalhm6PMi9rur:bhgx9hq)*
<!-- Sent from Reviewable.io -->

bradking commented:
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[drake/common/functional_form.h, line 17 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYnYEXVfH45T57o1ZE:-KUbQz9MaqZY5OTjcnKz:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L17)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> Please remove `@brief` (here and elsewhere). We switched to autobrief so Doxygen will pick up the first sentence as the brief anyway so we banned `@brief` as clutter.
</details>
Done.

---

*[drake/common/functional_form.h, line 23 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYorMqr2_9PhMRJ0MS:-KUbQtKrmllUM_QIWIqM:b-alg49p) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L23)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> Doxygen documentation says: `Anchors can currently only be put into a comment block that is marked as a page (using \page) or mainpage (\mainpage).` Does that apply here? If not, do these anchors actually work?
</details>
The anchors do work.  See example [here](http://drake.mit.edu/doxygen_cxx/classdrake_1_1_functional_form.html#differentiable).

---

*[drake/common/functional_form.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYnmH7VstFd6Kj-hqu:-KUbU8xCqQb-SE5g8jv6:b-3gt9of) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L42)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> BTW, some time ago Doxygen added Markdown support. That means you can use backtics for a tidier way of getting code fonts. So
> ```
> `f(x)` and `f` work
> ```
> Easier to read in the header file and produces the same doxygen output. Other Markdown goodies work too.
</details>
Thanks.  I've converted those that work with this syntax.  However, due to [this limitation](https://www.stack.nl/~dimitri/doxygen/manual/markdown.html#mddox_code_spans) not all of them work.  I've noted this in the commit message.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3853)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[drake/common/functional_form.h, line 23 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYorMqr2_9PhMRJ0MS:-KUbptRcWG9r1XnH2xOn:b-hvpa7t) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L23)):*
<details><summary><i>Previously, bradking (Brad King) wrote…</i></summary>
> The anchors do work.  See example [here](http://drake.mit.edu/doxygen_cxx/classdrake_1_1_functional_form.html#differentiable).
</details>
Oh, that's very nice. Thanks.

---

*[drake/common/functional_form.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-KUYnmH7VstFd6Kj-hqu:-KUbpzjLPXasChtKBvwM:blu0u6k) ([raw file](https://github.com/robotlocomotion/drake/blob/5e70be0c4315644cda873d78c0457acce0707968/drake/common/functional_form.h#L42)):*
<details><summary><i>Previously, bradking (Brad King) wrote…</i></summary>
> Thanks.  I've converted those that work with this syntax.  However, due to [this limitation](https://www.stack.nl/~dimitri/doxygen/manual/markdown.html#mddox_code_spans) not all of them work.  I've noted this in the commit message.
</details>
Rats! That's a shame -- I didn't know about that kludge.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3853#-:-KUbqFYS5w028q6zAyy5:bnfp4nl)*
<!-- Sent from Reviewable.io -->
